source/base/function_signed_distance.cc

```diff
@@ -353,6 +353,166 @@ namespace Functions
 
       return bounding_box.signed_distance(p);
     }
+
+
+
+    template <int dim>
+    ZalesakDisk<dim>::ZalesakDisk(const Point<dim> &center,
+                                  const double      radius,
+                                  const double      notch_width,
+                                  const double      notch_height)
+      : sphere(center, radius)
+      , notch(
+          {// bottom left
+           (dim == 1) ?
+             Point<dim>(center[0] - 0.5 * notch_width) :
+           (dim == 2) ?
+             Point<dim>(center[0] - 0.5 * notch_width,
+                        center[1] - std::sqrt(std::pow(radius, 2) -
+                                              std::pow(0.5 * notch_width, 2))) :
+             Point<dim>(center[0] - 0.5 * notch_width,
+                        std::numeric_limits<
+                          double>::lowest() /* notch is open in y-direction*/,
+                        center[2] - std::sqrt(std::pow(radius, 2) -
+                                              std::pow(0.5 * notch_width, 2))),
+           // top right
+           (dim == 1) ?
+             Point<dim>(center[0] + 0.5 * notch_width) :
+           (dim == 2) ?
+             Point<dim>(center[0] + 0.5 * notch_width,
+                        center[1] + notch_height - radius) :
+             Point<dim>(center[0] + 0.5 * notch_width,
+                        std::numeric_limits<
+                          double>::max() /* notch is open in y-direction*/,
+                        center[2] + notch_height - radius)})
+      , radius(radius)
+      , center_sphere(center)
+    {
+      AssertThrow(
+        notch_width <= 2 * radius,
+        ExcMessage(
+          "The width of the notch must be less than the circle diameter. Abort..."));
+      AssertThrow(
+        notch_height <= 2 * radius,
+        ExcMessage(
+          "The height of the notch must be less than the circle diameter. Abort..."));
```

Comment:
Let's just use regular `Assert` here and drop the last half-sentence (our error messages typically describe what is wrong -- a state -- rather than what will happen -- an action):
```suggestion
      Assert(
        notch_width <= 2 * radius,
        ExcMessage(
          "The width of the notch must be less than the circle diameter."));
      Assert(
        notch_height <= 2 * radius,
        ExcMessage(
          "The height of the notch must be less than the circle diameter."));
```